src/gui/explorer.go

```diff
@@ -104,31 +106,43 @@ func coinSupply(gateway *daemon.Gateway, w http.ResponseWriter, r *http.Request)
 		return nil
 	}
 
-	// skip locked addresses for total coinhour calculation
+	// locked distribution addresses
 	lockedAddrs := visor.GetLockedDistributionAddresses()
-	var filterInlocked []daemon.OutputsFilter
-	filterInlocked = append(filterInlocked, daemon.FbyAddressesNotIncluded(lockedAddrs))
-	lockedOutputs, err := gateway.GetUnspentOutputs(filterInlocked...)
 
-	// Get total coin hours of confirmed unspents
+	// get total coins hours which excludes locked distribution addresses
+	var coinHourFilter []daemon.OutputsFilter
+	coinHourFilter = append(coinHourFilter, daemon.FbyAddressesNotIncluded(lockedAddrs))
+	lockedOutputs, err := gateway.GetUnspentOutputs(coinHourFilter...)
+
+	// Get coin hours from confirmed unspents only
 	var totalCoinHours uint64
 	for _, out := range lockedOutputs.HeadOutputs {
 		totalCoinHours += out.Hours
 	}
 
+	// get current coin hours which excludes all distribution addresses
+	coinHourFilter = append(coinHourFilter, daemon.FbyAddressesNotIncluded(unlockedAddrs))
+	normalOutputs, err := gateway.GetUnspentOutputs(coinHourFilter...)
```

Comment:
Is it not possible to apply the filters ourselves? It's querying the full database 3 times in this endpoint now